File: React/ShoppingList/ShoppingList.Server/Models/ShoppingListModel.cs
namespace ShoppingList.Server.Models;

public class ShoppingListModel
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public List<ItemModel> Items { get; set; } = [];
    public DateTime CreatedAt { get; set; }
    public DateTime ModifiedAt { get; set; }
}

File: React/ShoppingList/shoppinglist.server.express/src/models/shoppinglist.model.js
import mongoose from "mongoose";

const shoppingListSchema = new mongoose.Schema({
    name: String,
    createdAt: String,
    items: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: "Item",
        }
    ],
}, {
    timestamps: true,
});

shoppingListSchema.set('toJSON', {
    transform: (doc, ret) => {
        const {_id, __v, updatedAt, ...cleanShoppingList} = ret;
        
        return {
            ...cleanShoppingList,
            id: _id.toString(),
            modifiedAt: updatedAt,
        };
    }
});

export default mongoose.model('ShoppingList', shoppingListSchema);